packages/js/src/ui/components/primitives/Tabs/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,61 @@
+import { Accessor, createEffect, createSignal, onCleanup, Setter } from 'solid-js';
+
+export const useKeyboardNavigation = ({
+  activeTab,
+  setActiveTab,
+  tabsContainer,
+}: {
+  activeTab: Accessor<string>;
+  setActiveTab: Setter<string>;
+  tabsContainer: Accessor<HTMLDivElement | undefined>;
+}) => {
+  const [keyboardNavigation, setKeyboardNavigation] = createSignal(false);
+
+  createEffect(() => {
+    const handleTabKey = (event: KeyboardEvent) => {
+      if (event.key !== 'Tab') {
+        return;
+      }
+
+      const tabs = tabsContainer()?.querySelectorAll('[role="tab"]');
+      if (!tabs || !document.activeElement) {
+        return;
+      }
+
+      setKeyboardNavigation(Array.from(tabs).includes(document.activeElement));
+    };
+
+    document.addEventListener('keyup', handleTabKey);
+
+    return onCleanup(() => document.removeEventListener('keyup', handleTabKey));
+  });
```

Comment:
listen to the `tab` keyboard events and check if `document.activeElement` is the "tab dom element", if yes allow keyboard navigation with `ArrowLeft` or `ArrowRight`